holoviews/element/graphs.py

```diff
@@ -392,3 +392,119 @@ class EdgePaths(Path):
     """
 
     group = param.String(default='EdgePaths', constant=True)
+
+
+class TriMesh(Graph):
+    """
+    A TriMesh represents a mesh of triangles represented as the
+    simplices and nodes. The simplices represent a indices into the
+    nodes array. The mesh therefore follows a datastructure very
+    similar to a graph, with the abstract connectivity between nodes
+    stored on the TriMesh element itself, the node positions stored on
+    a Nodes element and the concrete paths making up each triangle
+    generated when required by accessing the edgepaths.
+
+    Unlike a Graph each simplex is represented as the node indices of
+    the three corners of each triangle.
+    """
+
+    kdims = param.List(default=['node1', 'node2', 'node3'],
+                       bounds=(3, 3), doc="""
+        Dimensions declaring the node indices of each triangle.""")
+
+    group = param.String(default='TriMesh', constant=True)
+
+    _node_type = Nodes
+
+    _edge_type = EdgePaths
+
+    def __init__(self, data, kdims=None, vdims=None, **params):
+        if isinstance(data, tuple):
+            data = data + (None,)*(3-len(data))
+            edges, nodes, edgepaths = data
+        else:
+            edges, nodes, edgepaths = data, None, None
+        if nodes is None:
+            if isinstance(edges, list) and len(edges) == 0:
+                nodes = []
+            else:
+                raise ValueError("TriMesh expects both simplices and nodes "
+                                 "to be supplied.")
+
+        if isinstance(nodes, self._node_type):
+            pass
+        elif isinstance(nodes, Points):
+            # Add index to make it a valid Nodes object
+            nodes = self._node_type(Dataset(nodes).add_dimension('index', 2, np.arange(len(nodes))))
+        elif not isinstance(nodes, Dataset) or nodes.ndims in [2, 3]:
+            try:
+                # Try assuming data contains indices (3 columns)
+                nodes = self._node_type(nodes)
+            except:
+                # Try assuming data contains just coordinates (2 columns)
+                try:
+                    points = Points(nodes)
+                    ds = Dataset(points).add_dimension('index', 2, np.arange(len(points)))
+                    nodes = self._node_type(ds)
+                except:
+                    raise ValueError("Nodes argument could not be interpreted, expected "
+                                     "data with two or three columns representing the "
+                                     "x/y positions and optionally the node indices.")
+        if edgepaths is not None and not isinstance(edgepaths, self._edge_type):
+            edgepaths = self._edge_type(edgepaths)
+        super(TriMesh, self).__init__(edges, kdims=kdims, vdims=vdims, **params)
+        self._nodes = nodes
+        self._edgepaths = edgepaths
+
+    @classmethod
+    def from_points(cls, points):
```

Comment:
Shouldn't this be ``from_nodes`` or ``from_vertices``?

``from_points`` doesn't sound like graph terminology...

Reply:
I think the fact that it is a Graph type is an implementation detail but I agree ``from_vertices`` is better.